packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -405,17 +426,40 @@ Future<void> _buildGradleProjectV2(
     command.add('-Ptarget-platform=${getNameForTargetPlatform(buildInfo.targetPlatform)}');
 
   command.add(assembleTask);
+  bool potentialAndroidXFailure = false;
   final int exitCode = await runCommandAndStreamOutput(
     command,
     workingDirectory: flutterProject.android.hostAppGradleRoot.path,
     allowReentrantFlutter: true,
     environment: _gradleEnv,
-    filter: logger.isVerbose ? null : ndkMessageFilter,
+    // Feed every line to our mapFunction, we want to branch some logic off of multiple possible patterns.
+    filter: RegExp('.*'),
```

Comment:
there's gotta be a better way to do that. At a minimum, the empty string is a pattern that matches everything.